outlines/testing-outline.md

```diff
@@ -0,0 +1,56 @@
+# Testing
+
+1. Testing your Meteor application
+  1. Basic intro to testing concepts
+  2. Unit testing to ensure a module works in isolation
+    1. Using stubs / mocks / spies to isolate a module
+  3. Integration testing to ensure parts of the system works together
+  4. End-to-end/acceptance testing to ensure the system works in totality
+  5. Load/stress testing to ensure the application can handle real world load
+2. Challenges of testing a Meteor application
+  1. The client/server divide, ensuring data is available on the client
+  2. Reactivity + testing the system responds to changing data properly
+3. Generating test data
+  1. How the test database is created/cleared
+  2. Defining factories using factory package
+  3. Stubbing collections to create test data client side
+4. Unit testing in Meteor: package testing
+  1. Right now a package is the best system we have for isolating code and running tests
+  2. Using practicalmeteor:mocha to run package tests with mocha
+    1. Using `beforeEach` / etc to stub/mock
+  3. Isolating
+    1. Other packages w/ sinon
+    2. Everything with https://github.com/meteor-velocity/meteor-stubs
+    3. Collections with stub collections
+    4. Publications with stub subscriptions
+    5. Methods? w/ ?
+    6. Other bits of Meteor (e.g. `Meteor.userId()`) -- on an adhoc basis?
+  4. Testing various kinds of things:
+    1. General utilities
+    2. Collections / Models
+    3. Methods -- how to simulate `userId`
```

Comment:
This should be easy with the methods package, since you might be able to call the body of the method independently